packages/classes/src/PayloadArray.js

```diff
@@ -0,0 +1,100 @@
+import Payload from './Payload'
+
+/**
+ * PayloadArray stores a series of Payloads in a data structure that allows for easy filtering
+ * and lookup by position and type.
+ */
+class PayloadArray {
+	payloads = []
+
+	constructor(...payloads) {
+		[...payloads].forEach((payload) => this.insert(payload))
```

Comment:
Not sure if the `[...payloads]` does anything here, because I believe it shallow copies and you're inserting the payloads into a new array anyway.

Reply:
Also, should this have a JSDoc?

Reply:
Lol why did I do that (probably copypasta somewhere)